File: app/src/main/java/com/codeprogression/mvpb/MainPresenter.java
package com.codeprogression.mvpb;

import com.codeprogression.mvpb.core.BaseBindingPresenter;
import com.codeprogression.mvpb.core.PerActivity;

import javax.inject.Inject;

@PerActivity
public class MainPresenter extends BaseBindingPresenter<MainView.ViewModel> {

    @Inject
    public MainPresenter() {
    }

    @Override
    protected void load() {
        if (getViewModel().number.get() < 10) {
            getViewModel().number.set(10);
        }

    }


    public void add(){
        int number = getViewModel().number.get() + 1;
        getViewModel().number.set(number);
    }

}

File: app/src/main/java/com/codeprogression/mvpb/MainPresenter.kt
package com.codeprogression.mvpb

import com.codeprogression.mvpb.core.BaseBindingPresenter
import com.codeprogression.mvpb.core.PerActivity

import javax.inject.Inject

@PerActivity
class MainPresenter
@Inject
constructor() : BaseBindingPresenter<MainView.ViewModel>() {

    override fun load() {
        if (viewModel.number.get() < 10) {
            viewModel.number.set(10)
        }
    }

    fun add() {
        val number = viewModel.number.get() + 1
        viewModel.number.set(number)
    }

}